tests/tests_lite/plugins/collectives/test_torch_.py

```diff
@@ -226,18 +234,21 @@ def test_collectives_distributed(n):
 
 
 def _test_two_groups(strategy, left_collective, right_collective):
-    left_collective.create_group(ranks=[0, 1])
-    right_collective.create_group(ranks=[1, 2])
+    with check_destroy_group():  # the fixture does not work inside the launched function, use it manually
+        left_collective.create_group(ranks=[0, 1])
+        right_collective.create_group(ranks=[1, 2])
 
-    if strategy.global_rank in (0, 1):
-        tensor = torch.tensor(strategy.global_rank)
-        left_collective.all_reduce(tensor)
-        assert tensor == 1
-    right_collective.barrier()
-    if right_collective.rank >= 0:
         tensor = torch.tensor(strategy.global_rank)
-        right_collective.all_reduce(tensor)
-        assert tensor == 3
+        if strategy.global_rank in (0, 1):
+            tensor = left_collective.all_reduce(tensor)
+            assert tensor == 1
+        right_collective.barrier()  # avoids deadlock for global rank 1
```

Comment:
the comment here is correct. the reason behind it is explained here in the warning in the pytorch docs in case you were wondering:
https://pytorch.org/docs/stable/distributed.html#groups

Reply:
Thank you! That reference is very useful. Although it says it applies to NCCL, when this test uses Gloo. Maybe this limitation applies to both and the docs just don't mention it